### PR TITLE
Correct tag on release

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

- Changes docker tag from `x.y.z-x.y.z` to `x.y.z`. 
- fixes Upgrade e2e test on release

medic/cht-core#7902
medic/cht-core#7901

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:7902-correct-tag/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:7902-correct-tag/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:7902-correct-tag/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ env:
   ECR_REPO: '720541322708.dkr.ecr.eu-west-2.amazonaws.com/medic'
   ECR_PUBLIC_REPO: 'public.ecr.aws/s5s3h4s7'
   COUCHDB_LOG_LEVEL: 'debug'
+  TAG: ${{ (github.ref_type == 'tag' && github.ref_name) || '' }}
```
Comment (garethbowen) [inline .github/workflows/build.yml:16]: If I'm reading this right, all the code in the `scripts` dir is correct, we just needed to pass in the TAG?

**Review by garethbowen — APPROVED** [review]
Nice and simple, thanks!

File: .github/workflows/build.yml (hunk 1 above)
Comment (dianabarsan) [reply .github/workflows/build.yml:16]: Yup, I think that at some point the `TAG` was populated by something (there were already references to this) but then it got removed in some update to GH builds or something. 

I tested this on a fork of mine: 
https://github.com/dianabarsan/cht-core/actions/runs/3640208802/jobs/6144573769#step:7:20 

I just added a build job that prints the result of `getVersion` and pushed a tag. The output was the tag name itself.

**Comment by dianabarsan** [comment]
@garethbowen I realised I had missed something, where the test build when a tag was pushed would have tagged the test version with the final tag. Fixed that in an additional commit.

**Pull request merged**